pkg/usersignup/usersignup.go

```diff
@@ -0,0 +1,84 @@
+package usersignup
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+	"strings"
+
+	toolchainv1alpha1 "github.com/codeready-toolchain/api/pkg/apis/toolchain/v1alpha1"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/validation"
+)
+
+var (
+	specialCharRegexp = regexp.MustCompile("[^A-Za-z0-9]")
+	onlyNumbers       = regexp.MustCompile("^[0-9]*$")
+)
+
+func GenerateCompliantUsername(instance *toolchainv1alpha1.UserSignup, cl client.Client, forbiddenUsernamePrefixes []string ) (string, error) {
```

Comment:
Correct me if I'm wrong, but this should not be here. We wanted to move only the tranformation of the name - just `TransformUsername` function, not check of the duplications. This gets already existing MURs and appends only the number suffix (if there is a name conflict), so it should not have any significant impact on the resulting name format.
In addition, at the time of the UserSignup creation, it's still not clear if any suffix will be added or not - this is evaluated when the UserSignup is approved, which can take some time.

Reply:
Ohhh okay, sorry, i moved both, I'll move GenerateCompliantUsername back in that case..

Reply:
@MatousJobanek fixed here: c34f4b515b6cfe887adf2a55f09ed4b8a26e9b8c & 8325ca1002576ab4a71f891d0fdedc548580fb8f